spec/mainadditions.html

```diff
@@ -271,6 +271,247 @@ <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Record
+
+      <emu-alg>
+        1. If the implementation only supports the UTC time zone, return *"UTC"*.
+        1. Let _systemTimeZoneString_ be the String representing the host environment's current time zone, either a primary time zone identifier or <del>an offset time zone identifier</del><ins>a |NormalizedUTCOffset|</ins>.
+        1. If IsTimeZoneOffsetString(_systemTimeZoneString_) is *true*, return _systemTimeZoneString_.
+        1. Return _systemTimeZoneString_.
+      </emu-alg>
+
+      <emu-note>
+        <p>
+          To ensure the level of functionality that implementations commonly provide in the methods of the Date object, it is recommended that SystemTimeZoneIdentifier return an IANA time zone name corresponding to the host environment's time zone setting, if such a thing exists.
+          GetNamedTimeZoneEpochNanoseconds and GetNamedTimeZoneOffsetNanoseconds must reflect the local political rules for standard time and daylight saving time in that time zone, if such rules exist.
+        </p>
+        <p>For example, if the host environment is a browser on a system where the user has chosen US Eastern Time as their time zone, SystemTimeZoneIdentifier returns *"America/New_York"*.</p>
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-time-zone-offset-strings">
+      <h1>Time Zone Offset String <del>Format</del><ins>Formats</ins></h1>
+
+      <del class="block">
+      <p>
+        ECMAScript defines a string interchange format for UTC offsets, derived from ISO 8601.
+        The format is described by the following grammar.
+        The usage of Unicode code points in this grammar is listed in <emu-xref href="#table-time-zone-offset-string-code-points"></emu-xref>.
+      </p>
+      </del>
+
+      <ins class="block">
+      <p>
+        ECMAScript defines string interchange formats for UTC offsets, derived from ISO 8601.
+        UTC offsets that represent offset time zone identifiers, or that are intended for interoperability with ISO 8601, use only hours and minutes and are specified by |UTCOffsetMinutePrecision|.
+        UTC offsets that represent the offset of a named or custom time zone can be more precise, and are specified by |UTCOffsetNanosecondPrecision|.
+      </p>
+      <p>
+        These formats are described by the ISO String grammar in <emu-xref href="#sec-temporal-iso8601grammar"></emu-xref>.
+        The usage of Unicode code points in this grammar is listed in <emu-xref href="#table-time-zone-offset-string-code-points"></emu-xref>.
+      </p>
+      </ins>
+
+      <p>[...]</p>
+
+      <emu-note type="editor">
+        The grammar in this section should be deleted; it is replaced by the ISO 8601 String grammar in <emu-xref href="#sec-temporal-iso8601grammar"></emu-xref>.
+      </emu-note>
+
+      <p>[...]</p>
+
+      <ins class="block">
+      <emu-clause id="sec-interprettimezoneoffsetparseresult" type="abstract operation">
+        <h1>
+          InterpretTimeZoneOffsetParseResult (
```

Comment:
I'd also be happier to see this integrated into a single Record-returning ParseTimeZoneOffsetString.